libs/api/api_internal.h

```diff
@@ -607,64 +606,38 @@ ebpf_object_load(_Inout_ struct bpf_object* object) noexcept;
 
-typedef int (*ring_buffer_sample_fn)(void* ctx, void* data, size_t size);
-
 /**
- * @brief Subscribe for notifications from the input ring buffer map.
+ * @brief Subscribe for notifications from the input perf event array or a ring buffer map.
  *
- * @param[in] ring_buffer_map_fd File descriptor to the ring buffer map.
- * @param[in, out] sample_callback_context Pointer to supplied context to be passed in notification callback.
- * @param[in] sample_callback Function pointer to notification handler.
- * @param[out] subscription Opaque pointer to ring buffer subscription object.
- *
- * @retval EBPF_SUCCESS The operation was successful.
- * @retval EBPF_NO_MEMORY Out of memory.
- */
-_Must_inspect_result_ ebpf_result_t
-ebpf_ring_buffer_map_subscribe(
-    fd_t ring_buffer_map_fd,
-    _Inout_opt_ void* sample_callback_context,
-    ring_buffer_sample_fn sample_callback,
-    _Outptr_ ring_buffer_subscription_t** subscription) noexcept;
-
-/**
- * @brief Unsubscribe from the ring buffer map event notifications.
- *
- * @param[in] subscription Pointer to ring buffer subscription to be canceled.
- */
-bool
-ebpf_ring_buffer_map_unsubscribe(_In_ _Post_invalid_ ring_buffer_subscription_t* subscription) noexcept;
-
-typedef void (*perf_buffer_sample_fn)(void* ctx, int cpu, void* data, uint32_t size);
-typedef void (*perf_buffer_lost_fn)(void* ctx, int cpu, uint64_t cnt);
-
-/**
- * @brief Subscribe for notifications from the input perf event array map.
- *
- * @param[in] perf_event_array_map_fd File descriptor to the perf event array map.
- * @param[in, out] callback_context Pointer to supplied context to be passed in notification callback.
+ * @param[in] map_fd File descriptor to the perf event array or a ring buffer map.
+ * @param[in] cpu_ids The CPU Ids corresponding to this subscription. For a ring buffer map this is a single value with
+ * Id 0.
+ * @param[in] cpu_id_count The count of the elements in the cpu_ids parameter.
+ * @param[in] callback_context Pointer to supplied context to be passed in notification callback.
  * @param[in] sample_callback Function pointer to notification handler.
  * @param[in] lost_callback Function pointer to lost record notification handler.
- * @param[out] subscription Opaque pointer to perf event array subscription object.
+ * @param[out] subscription Opaque pointer to the subscription object.
  *
  * @retval EBPF_SUCCESS The operation was successful.
  * @retval EBPF_NO_MEMORY Out of memory.
  */
 _Must_inspect_result_ ebpf_result_t
-ebpf_perf_event_array_map_subscribe(
-    fd_t perf_event_array_map_fd,
+ebpf_map_subscribe(
+    fd_t map_fd,
+    _In_reads_(cpu_id_count) uint32_t* cpu_ids,
+    _In_ size_t cpu_id_count,
     _Inout_opt_ void* callback_context,
-    perf_buffer_sample_fn sample_callback,
-    perf_buffer_lost_fn lost_callback,
-    _Outptr_ perf_event_array_subscription_t** subscription) noexcept;
+    _In_ const void* sample_callback,
```

Comment:
nit: should this be `perf_buffer_sample_fn` instead of void*? same for `lost_callback`?